controllers/cluster_controller_test.go

```diff
@@ -2547,158 +2579,53 @@ var _ = Describe("cluster_controller", func() {
 		When("creating a cluster with RocksDB as storage engine", func() {
 			When("using rocksdb-v1 engine", func() {
 				When("using 6.3.26", func() {
-					BeforeEach(func() {
-						cluster.Spec.DatabaseConfiguration.StorageEngine = fdbv1beta2.StorageEngineRocksDbV1
-						cluster.Spec.Version = "6.3.26"
-						err := k8sClient.Update(context.TODO(), cluster)
-						Expect(err).NotTo(HaveOccurred())
-						shouldCompleteReconciliation = false
-					})
-					It("generations are not matching", func() {
-						generations, err := reloadClusterGenerations(cluster)
-						Expect(err).NotTo(HaveOccurred())
-						Expect(generations.Reconciled).ToNot(Equal(cluster.ObjectMeta.Generation))
-					})
+					BeforeEach(updateStorageEngineVersionIncompleteReconcilliation(cluster, fdbv1beta2.StorageEngineRocksDbV1, "6.3.26", &shouldCompleteReconciliation))
+					It("generations are not matching", generationsNotMatching(cluster))
```

Comment:
```suggestion
					BeforeEach(fund() {
					     updateStorageEngineVersionIncompleteReconcilliation(cluster, fdbv1beta2.StorageEngineRocksDbV1, "6.3.26", &shouldCompleteReconciliation)
					})
					It("generations are not matching", func() {
					    generationsNotMatching(cluster)}
					)
```

Ginkgo doesn't work like this and requires to have a closure passed in. Otherwise the variables are not substituted and the cluster will be nil.